### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by arielofer**
this pull request was opened later then "class updates" but needs to be closed before it

### CODE REVIEW

**Review by agamm — COMMENTED** [review]
See comments on other PR (I saw it first and put all of the comments there). Change here and I'll approve this first.

File: gamebook/nextscene.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class Decision(object):
+class NextScene(object):
```
Comment (agamm) [inline gamebook/nextscene.py:1]: 👍

File: gamebook/options.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import random
 
 class Options():
-    """an option which the player won't necessarily be succesful in preforming(depends on the decision's success rate)"""
+    """an option which the player won't necessarily be succesful in preforming
```
Comment (agamm) [inline gamebook/options.py]: "A set of options...."

File: gamebook/scene.py [hunk 3]
```diff
@@ -3,28 +3,28 @@ class Scene(object):
         arguments:
         desc - A string containing what is hppening in this current scene
 
-        choices - a list of options that the user can choose to continue to the next scene
+        options - a list of options that the user can choose to continue to the next scene
```
Comment (agamm) [inline gamebook/scene.py]: can choose **.** " 
(emove later part of comment)

File: gamebook/tests/test_nextscene.py [hunk 4]
```diff
@@ -1,17 +1,10 @@
...
-        next_scene: scene1
-        success_rate: 1.0 (default)
-    
-"""
-def test_initialization():
-
-    assert nextscene1.get_scene() == scene1
-    assert nextscene1.get_scene_name() == scene1.get_name()
-    assert nextscene1.get_success_rate() == 1.0
+""" scene1: name = "scene1" """
```
Comment (agamm) [inline gamebook/tests/test_nextscene.py]: remove

File: gamebook/tests/test_nextscene.py [hunk 5]
```diff
@@ -1,17 +1,10 @@
-from gamebook.tests.mock_scenes import scene1
-from gamebook.tests.mock_nextscene import nextscene1
+from gamebook.nextscene import NextScene
+from gamebook.scene import Scene
+from gamebook.tests.test_scene import scene1
 
-"""
-    using mock Next_scene nextscene1:
+nextscene1 = NextScene(scene1, 1.0)
```
Comment (agamm) [inline gamebook/tests/test_nextscene.py]: move into unit test, it should be self contained.

File: gamebook/tests/test_scene.py [hunk 6]
```diff
@@ -1,29 +1,20 @@
...
+dead_scene = Scene("death_scene", "game over", [])
 
-scene1:
-    name: scene1
-    desc: welcome to scene no 1
-    choices:[
-            Option('Go left', key=['l', 'left'], nextscene=[Next_scene(dead_scene)]),
-            Option('Go right', key=['r', 'right'], nextscene=[Next_scene(dead_scene)])
-            ]
+scene1 = Scene("scene1", "At last your two-day hike is over.", [
```
Comment (agamm) [inline gamebook/tests/test_scene.py]: same as nextscene1's comment

File: gamebook/tests/test_scene.py [hunk 7]
```diff
@@ -1,29 +1,20 @@
-from gamebook.tests.mock_scenes import scene1, dead_scene
+from gamebook.scene import Scene
+from gamebook.nextscene import NextScene
+from gamebook.option import Option
 
-"""
-using mock scenes scene1 and dead_scene
+dead_scene = Scene("death_scene", "game over", [])
```
Comment (agamm) [inline gamebook/tests/test_scene.py]: same as nextscene1's comment

File: gamebook/tests/test_trait.py [hunk 8]
```diff
@@ -1,21 +1,15 @@
...
-
-trait1:
-    name: trait1
-    const: 6
-    dice_num: not set - checking if default is 0
-
-trait2:
-    dice_num: 1
-"""
+trait1 = Trait("trait1", 6, dice_num=3)
```
Comment (agamm) [inline gamebook/tests/test_trait.py]: same as nextscene1's comment

**Review by agamm — CHANGES_REQUESTED** [review]
Looks much better, apply changes and let's merge.

File: gamebook/tests/test_nextscene.py [hunk 9]
```diff
@@ -2,9 +2,7 @@
 from gamebook.scene import Scene
 from gamebook.tests.test_scene import scene1
 
-nextscene1 = NextScene(scene1, 1.0)
-
-""" scene1: name = "scene1" """
-
 def test_get_scene_name():
+    nextscene1 = NextScene(scene1, 1.0)
```
Comment (agamm) [inline gamebook/tests/test_nextscene.py:6]: add scene1 to the test for it to be more understandable now.

**Review by agamm — COMMENTED** [review]
Minor comment, change then merge.

File: gamebook/tests/test_nextscene.py (hunk 9 above)
Comment (arielofer) [reply gamebook/tests/test_nextscene.py:6]: i already merged it :(

Comment (arielofer) [reply gamebook/tests/test_nextscene.py:6]: can i undo it?

Comment (agamm) [reply gamebook/tests/test_nextscene.py:6]: next PR.

Comment (arielofer) [reply gamebook/tests/test_nextscene.py:6]: :+1:

**Pull request merged**